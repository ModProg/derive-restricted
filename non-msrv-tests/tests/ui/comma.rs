#[derive(derive_where::DeriveWhere)]
#[derive_where(Clone,; T)]
struct Test1<T>(T);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Clone,,)]
struct Test2<T>(T);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Clone; T;)]
struct Test3<T>(T);

#[derive(derive_where::DeriveWhere)]
#[derive_where(Clone; T,,)]
struct Test4<T>(T);

fn main() {}
