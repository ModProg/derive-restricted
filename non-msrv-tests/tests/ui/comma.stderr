error: expected `,`
 --> tests/ui/comma.rs:2:24
  |
2 | #[derive_where(Clone; T;)]
  |                        ^

error: expected type to bind to, expected one of: `for`, parentheses, `fn`, `unsafe`, `extern`, identifier, `::`, `<`, square brackets, `*`, `&`, `!`, `impl`, `_`, lifetime
 --> tests/ui/comma.rs:6:25
  |
6 | #[derive_where(Clone; T,,)]
  |                         ^
